scripts/install.sh

```diff
@@ -0,0 +1,224 @@
+#!/bin/bash
+
+ROOT_DIR="$(pwd)"
+
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    OS="macOS"
+    NUMBER_OF_PROCESSORS="$(sysctl -n hw.ncpu)"
+fi
+
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+    OS="Linux"
+    NUMBER_OF_PROCESSORS="$(nproc)"
+fi
+
+GENERATOR="Unix Makefiles"
+
+SUBMODULE_DIR="${ROOT_DIR}/vendor"
+SUBMODULE_INCLUDE_DIR="${SUBMODULE_DIR}/include"
+SUBMODULE_LIB_DIR="${ROOT_DIR}/lib/${OS}"
+
+echo "Installing vulkan dependencies..."
+
+update_submodules() {
+    git submodule update --init ${SUBMODULE_DIR}/$1
+}
+
+mkdirp() { 
+    mkdir -p $1 
+}
+
+setup_volk() {
+    echo "Cloning Volk..."
+    update_submodules volk
+
+    mkdirp ${SUBMODULE_DIR}/include/volk
+
+    cp ${SUBMODULE_DIR}/volk/volk.h ${SUBMODULE_INCLUDE_DIR}/volk 
+    cp ${SUBMODULE_DIR}/volk/volk.c ${SUBMODULE_INCLUDE_DIR}/volk
+}
+
+setup_glfw() {
+    echo "Cloning glfw..."
+
+    update_submodules glfw
+
+    cd ${SUBMODULE_DIR}/glfw ; cmake -G "${GENERATOR}" . ; make -j${NUMBER_OF_PROCESSORS}
+
+    mkdirp ${SUBMODULE_LIB_DIR}
+
+    cp ${SUBMODULE_DIR}/glfw/src/libglfw3.a ${SUBMODULE_LIB_DIR}
+
+    cd ${ROOT_DIR}
+}
+
+setup_moltenVk() {
+    echo "Cloning MoltenVk..."
+
+    update_submodules MoltenVK
+    cd ${SUBMODULE_DIR}/MoltenVK ; ./fetchDependencies --macos --v-headers-root ${SUBMODULE_DIR}/Vulkan-Headers
+    make macos -j${NUMBER_OF_PROCESSORS}
+
+    mkdirp ${SUBMODULE_LIB_DIR}/icd.d
+    mkdirp ${SUBMODULE_INCLUDE_DIR}/vulkan
+
+    cp ${SUBMODULE_DIR}/MoltenVK/Package/Latest/MoltenVK/dylib/macOS/** ${SUBMODULE_LIB_DIR}/icd.d 
+    cp ${SUBMODULE_DIR}/MoltenVK/Package/Latest/MoltenVK/dylib/macOS/**.dylib ${SUBMODULE_LIB_DIR} 
+
+    cd ${ROOT_DIR}
+}
+
+setup_vulkan_headers() {
+    update_submodules Vulkan-Headers
+
+    cd ${SUBMODULE_DIR}/Vulkan-Headers ; git fetch --all --tags ; git checkout tags/v1.3.211
+    
+    mkdirp ${SUBMODULE_DIR}/Vulkan-Headers/build
+
+    cd ${SUBMODULE_DIR}/Vulkan-Headers/build ; cmake -DCMAKE_INSTALL_PREFIX=install -G "${GENERATOR}" ..
+    cd ${SUBMODULE_DIR}/Vulkan-Headers/build ; cmake --build . --target install
+
```

Comment:
possible to run `git`/`cmake` on a directory?

Reply:
Yes. I've updated all of the cases where we run a `cd` command